src/types/mod.rs

```diff
@@ -1207,7 +1207,7 @@ impl BlockItemSummary {
     ///
     /// Returns [`Upward::Unknown`] when encountering unfamiliar data from newer
     /// versions of the node to stay forward-compatible.
-    pub fn affected_contracts(&self) -> Upward<Vec<ContractAddress>> {
+    pub fn affected_contracts(&self) -> Upward<Vec<Upward<ContractAddress>>> {
```

Comment:
Double `Upward`? Should we not return
```suggestion
    pub fn affected_contracts(&self) -> Vec<Upward<ContractAddress>> {
```